src/js/libgeo/GeoOps.js

```diff
@@ -59,6 +59,26 @@ geoOps.FreeLine.updatePosition = function(el) {
     putStateComplexVector(param); // copy param
     el.homog = General.withUsage(param, "Line");
 };
+geoOps.FreeLine.getRandomMove = function(el) {
+    var rcomplex = function() {
+        var fact = 0.1;
+        return CSNumber.complex((Math.random() - 0.5) * fact, (Math.random() - 0.5) * fact);
+    };
+
+    var l = el.homog;
+    var tt = List.turnIntoCSList([l.value[0], l.value[1], CSNumber.zero]);
+    var p = List.turnIntoCSList([rcomplex(), rcomplex(), CSNumber.real(1)]);
+    var perp = List.cross(p, tt);
+
+    var move = List.cross(perp, l);
+    move = List.normalizeMax(move);
+
+    return {
+        type: "mouse",
+        value: move
+    };
+
+};
```

Comment:
I'm trying to understand this, and I think that some more verbose names and/or comments might help. `l` is the line as it is. `tt` is the point at infinity orthogonal to it. `p` is some random point near the origin. Now you connect `tt` and `p` to obtain a line `perp` which is perpendicular to the original `l`. Next you intersect `perp` with `l` and treat that intersection like a mouse move. The goal here is finding a small but definitely non-zero change, right? So the fact that you intersect with the line will ensure that the resulting point won't be too far away from the line. And the fact that the line `perp` passes close to the origin will make sure that the resulting point is again close to the origin, and thus to the point where the mouse would define the line in question. Did I understand this correctly?

If `l` is the line at infinity, then `tt`, `perp` and `move` will be zero. While this couldn't happen using the current mouse user interface, it could well happen using scripted movement, or perhaps some future input methods.